### ISSUE TITLE
Feature Request: Some sort of ARIA-landmarking for each header/section

### ISSUE THREAD

**Issue by lkarvec** [react-querybuilder/react-querybuilder#512]
It would be super cool to get better ARIA screenreader strings on the headers to delineate where a user's context is at on the query.

There's probably a way to do it like "if the user enters this section, read the header aria text out" but I'm not entirely sure.

I would assume the best way to read it out would be "You are in Query Builder - Section X" and then add "in Section Y" at the end of this string if it is nested.  You might be able to get away with an aria label but I would be inclined to landmark I thiiiiink.

**Comment by jakeboone02** [comment]
Just getting back to this and wanted to jot down some notes:

Going by https://www.w3.org/WAI/GL/wiki/Using_ARIA_landmarks_to_identify_regions_of_a_page, it seems like we would only need to add `role="form"` to the outermost `div`, and `aria-label="[something about position in the hierarchy]"` to each rule group. Both should be configurable, maybe within the `translations` prop.

According to https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/form.html:

> If a page includes more than one form landmark, each should have a unique label.

This complicates the `aria-label` config since it could be dynamic based on the `path`. Maybe it could accept a function with `path` as a parameter?

**Comment by jakeboone02** [comment]
Hey @lkarvec, I've been noodling and researching this one for a while and came up with a new prop (see #580) that generates a `title` for the outer `<div>` of each rule group. I was going to use `aria-label`, but after reading the ["first rule of ARIA"](https://www.w3.org/TR/using-aria/#rule1) (also on [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)) and testing with React Testing Library's [`toHaveAccessibleName`](https://github.com/testing-library/jest-dom#tohaveaccessiblename) matcher, it seems `title` is preferred.

If this doesn't work, please let me know if you have any suggestions. I don't have much (read: enough) experience with a11y.

**Comment by lkarvec** [comment]
I'm always stumbling through it myself, not gunna lie.  If you want some feedback in testing, I could ask our testing team to tackle it if free.  I do feel like this is adequate though.  Thank you!

**Comment by jakeboone02** [comment]
> I'm always stumbling through it myself, not gunna lie.  If you want some feedback in testing, I could ask our testing team to tackle it if free.  I do feel like this is adequate though.  Thank you!

Thanks for getting back to me. I would love for someone to take a look at it if possible. I'll see if I can publish an alpha version this weekend that you can use. It'll be the first v7 prerelease.

**Comment by jakeboone02** [comment]
I went ahead and merged #580 into `main`, which automatically closed this one, but I'd still like any feedback you or your testing team could provide. I also published [v7.0.0-alpha.0](https://www.npmjs.com/package/react-querybuilder/v/7.0.0-alpha.0) that includes the new `accessibleDescriptionGenerator` prop. The default implementation is [here](https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/generateAccessibleDescription.ts).
